mget.go

```diff
@@ -56,64 +79,58 @@ func (m *MGetRequest) SetIndex(index string) *MGetRequest {
 //   - The request to OpenSearch fails
 //   - The results json cannot be unmarshalled
 func (m *MGetRequest) Do(ctx context.Context, client *opensearch.Client) (*MGetResponse, error) {
-    bodyBytes, jErr := json.Marshal(m)
-    if jErr != nil {
-        return nil, jErr
-    }
-
-    osResp, rErr := opensearchapi.MgetRequest{
-        Index: m.Index,
-        Body:  bytes.NewReader(bodyBytes),
-    }.Do(ctx, client)
-
-    if rErr != nil {
-        return nil, rErr
-    }
-
-    resp := &MGetResponse{
-        StatusCode: osResp.StatusCode,
-        Header:     osResp.Header,
-    }
-
-    var respBuf bytes.Buffer
-    if _, err := respBuf.ReadFrom(osResp.Body); err != nil {
-        return nil, err
-    }
-
-    if err := json.Unmarshal(respBuf.Bytes(), &resp); err != nil {
-        return nil, err
-    }
-
-    return resp, nil
-}
-
-// mgetDoc is a simple struct representing an individual document to be fetched in a MultiGet request
-type mgetDoc struct {
-    ID    string `json:"_id,omitempty"`
-    Index string `json:"_index,omitempty"`
+	bodyBytes, jErr := json.Marshal(m.Source())
```

Comment:
It would be more idiomatic IMO to have `MGetRequest` implement `json.Marshaler`.  The logic would be the same, but this call would be `bodyBytes, jErr := json.Marshal(m)`
```
func (m *MGetRequest) MarshalJSON() ([]byte, error) {
   // same logic that you have in Source() to build the source map
   return json.Marshal(source)
}
```
This way we can even write tests to verify that the generated JSON matches what OpenSearch expects.